chain-extensions/unified-accounts/src/lib.rs

```diff
@@ -58,8 +63,17 @@ where
 
                 let base_weight = <T as frame_system::Config>::DbWeight::get().reads(1);
                 env.charge_weight(base_weight)?;
+
+                // read the storage item
+                let mapped = NativeToEvm::<T>::get(account_id.clone());
```

Comment:
`UA::to_h160` could be used to avoid relying on impl details directly.

Reply:
Agreed!